cmd/libsnap-confine-private/snap.c

```diff
@@ -168,3 +168,24 @@ void sc_snap_name_validate(const char *snap_name, struct sc_error **errorp)
  out:
 	sc_error_forward(errorp, err);
 }
+
+void sc_snap_drop_instance_name(const char *snap_name, char *base, size_t base_len)
+{
+	if (snap_name == NULL || base == NULL) {
+		die("base or snap name unset");
+	}
+
+	const char *pos = strchr(snap_name, '_');
+	size_t blen = 0;
+	if (pos == NULL) {
+		blen = strlen(snap_name);
+	} else {
+		blen = (pos - snap_name);
+	}
+
+	if (blen > base_len) {
+		die("base buffer too small");
+	}
+
+	memcpy(base, snap_name, blen);
```

Comment:
After correcting the above length check, please add nul termination: `base[blen] = '\0';`. You might be thinking the caller will initial 'base' properly, but in case it doesn't, this is super-cheap and safe.

Eg:
```
if (blen >= base_len) {
...
}
memcpy(base, snap_name, blen);
base[blen] = '\0';
```